diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index 433313a92..82c0870bd 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -391,7 +391,15 @@ CFG_NSEC_DDR_1_SIZE ?= 0x40000000UL
 endif
 
 ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mnevk))
+ifneq (,$(filter $(PEW_BOARD_TYPE),imx8mnpewmon))
+CFG_DDR_SIZE ?= 0x40000000
+else ifneq (,$(filter $(PEW_BOARD_TYPE),imx8mnpewgw))
+CFG_DDR_SIZE ?= 0x20000000
+else ifneq (,$(filter $(PEW_BOARD_TYPE),imx8mnpewbizgw))
+CFG_DDR_SIZE ?= 0x20000000
+else
 CFG_DDR_SIZE ?= 0x80000000
+endif
 CFG_UART_BASE ?= UART2_BASE
 endif
 
